classes/order/OrderHistory.php

```diff
@@ -372,32 +372,34 @@ public function changeIdOrderState($new_order_state, $id_order, $use_existing_pa
 
             foreach ($invoices as $invoice) {
                 /** @var OrderInvoice $invoice */
-                $rest_paid = $invoice->getRestPaid();
-                if ($rest_paid > 0) {
-                    $payment = new OrderPayment();
-                    $payment->order_reference = Tools::substr($order->reference, 0, 9);
-                    $payment->id_currency = $order->id_currency;
-                    $payment->amount = $rest_paid;
-
-                    if ($order->total_paid != 0) {
-                        $payment->payment_method = $payment_method->displayName;
-                    } else {
-                        $payment->payment_method = null;
-                    }
+                if ((int) $invoice->number > 0) {
```

Comment:
@Prestaworks For avoiding to complexify the loop, you can use `continue` for skipping the process and continue to process the list of invoices.
```suggestion
                if ((int) $invoice->number == 0) {
                    continue;
                }
```

Reply:
should it be
if (0 == (int) $invoice->number) {
                    continue;
                }
for following standards?

Reply:
Exactly ;) YodaStyle ;)

Reply:
It's modified.